google-cloud-spanner/src/main/java/com/google/cloud/spanner/SpannerExceptionFactory.java

```diff
@@ -257,35 +256,8 @@ private static boolean hasCauseMatching(
   }
 
   private static class Matchers {
-    static final Predicate<Throwable> isRetryableInternalError =
```

Comment:
I extracted these predicates into their own classes that are now exposed to the `PartitionedDmlTransaction`. This was the cleanest / lowest impact way I could find to de-duplicate the EOS retryable logic.